pennylane/workflow/jacobian_products.py

```diff
@@ -339,11 +339,11 @@ class DeviceDerivatives(JacobianProductCalculator):
 
     Args:
 
-        device (Union[pennylane.Device, pennylane.devices.Device]): the device for execution and derivatives.
+        device (Union[pennylane.devices.LegacyDevice, pennylane.devices.Device]): the device for execution and derivatives.
             Must support first order gradients with the requested configuration.
```

Comment:
```suggestion
        device (pennylane.devices.Device): the device for execution and derivatives.
            Must support first order gradients with the requested configuration.
```

Sorry I missed this one before. These classes only support the new device interface now.